astropy/units/format/ogip.py

```diff
@@ -331,11 +332,6 @@ def p_error(p):
 
         return parsing.yacc(tabmodule="ogip_parsetab", package="astropy/units")
 
-    @classmethod
-    def _parse_unit(cls, unit: str, detailed_exception: bool = True) -> UnitBase:
-        cls._validate_unit(unit, detailed_exception=detailed_exception)
-        return cls._units[unit]
```

Comment:
`OGIP` can now inherit the implementation from `FITS`, which is identical: https://github.com/astropy/astropy/blob/02c8eff5751e2677ca06a0867a0e4de7b39ecf50/astropy/units/format/fits.py#L64-L68

There are several simplifications that this pull request enables, but I only included this one because the others not as trivial.